planet/src/main/webapp/places.js

```diff
@@ -353,10 +353,21 @@ function renderResult(placeInfo) {
     document.getElementById('greeting').innerHTML = 'Find a location: ' + document.getElementById('pac-input').value;
 }
 
+/** Create and add save icon */
 function createIcon(placeID) {
-    // Create and add save icon 
     let icon = document.createElement('i');
-    icon.innerHTML = 'favorite_border';
+    let userID = currentUser.uid; 
+    
+    database.ref('users/' + userID +'/places').child(placeID).once('value').then(function(snapshot) {
+        // Check if place is already saved in database
```

Comment:
This comment is just repeating the code. It doesn't help the user understand.

Instead, we can explain _why_ the fact that it's in the database affects the styling.

E.g.
// Places are in the database if and only if they've been favourited by the user.